starfish/pipeline/filter/gaussian_low_pass.py

```diff
@@ -29,41 +32,44 @@ def __init__(self, sigma, is_volume: bool=False, verbose=False, **kwargs) -> Non
         self.verbose = verbose
 
     @classmethod
-    def get_algorithm_name(cls):
+    def get_algorithm_name(cls) -> str:
         return "gaussian_low_pass"
 
     @classmethod
-    def add_arguments(cls, group_parser):
+    def add_arguments(cls, group_parser: argparse.ArgumentParser) -> None:
         group_parser.add_argument(
-            "--sigma", default=1, type=int, help="standard deviation of gaussian kernel")
+            "--sigma", default=1, type=float, help="standard deviation of gaussian kernel")
 
     @staticmethod
-    def low_pass(image, sigma) -> numpy.ndarray:
-        """Apply a Gaussian blur operation over a multi-dimensional image.
+    def low_pass(image: np.ndarray, sigma: Union[Number, Tuple[Number]]) -> np.ndarray:
+        """
+        Apply a Gaussian blur operation over a multi-dimensional image.
 
         Parameters
         ----------
         image : np.ndarray
-            Image data
-        sigma : Union[float, int, Tuple]
+            2-d or 3-d image data
+        sigma : Union[Number, Tuple[Number]]
             Standard deviation of the Gaussian kernel that will be applied. If a float, an isotropic kernel will be
             assumed, otherwise the dimensions of the kernel give (z, x, y)
 
         Returns
         -------
         np.ndarray :
-            Blurred data in same shape as input image.
+            Blurred data in same shape as input image, converted to np.uint16 dtype.
```

Comment:
Should we constrain the input types? At the very least, I would imagine that the skimage limitations should apply here.

Reply:
There's some craziness here. I'm just documenting it in this PR but I've added a todo to look into why this is here and hopefully remove it.

Reply:
We can safely cast the image types, added a warning when that happens.